x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/process/AbstractNativeProcessTests.java

```diff
@@ -88,7 +88,6 @@ public void testStart_DoNotDetectCrashWhenNoInputPipeProvided() throws Exception
         when(processPipes.getProcessInStream()).thenReturn(Optional.empty());
         try (AbstractNativeProcess process = new TestNativeProcess()) {
             process.start(executorService);
-        } finally {
             mockNativeProcessLoggingStreamEnds.countDown();
             // Not detecting a crash is confirmed in terminateExecutorService()
         }
```

Comment:
I think this could be made even better by nesting the `finally` so that we count down before close whether the test passes or fails.

```
        try (AbstractNativeProcess process = new TestNativeProcess()) {
            try {
                process.start(executorService);
            } finally {
                mockNativeProcessLoggingStreamEnds.countDown();
                // Not detecting a crash is confirmed in terminateExecutorService()
            }
        }
```

(And obviously the same pattern for all the other places too.)

Or to avoid nested `try` blocks it could be done like we're already doing it in `testStart_DoNotDetectCrashWhenProcessIsBeingKilled`:

```
        AbstractNativeProcess process = new TestNativeProcess()) {
        try {
            process.start(executorService);
        } finally {
            mockNativeProcessLoggingStreamEnds.countDown();
            // Not detecting a crash is confirmed in terminateExecutorService()
            process.close();
        }
```

Reply:
Yes if the test fails or throws before `mockNativeProcessLoggingStreamEnds.countDown();` the 5s wait will be hit again. I've optimised for the common case and the fix was pleasingly simple so I didn't go any further.

Reply:
If you can't be bothered then let me push a change while the PR is open, as at least that will save the effort of two PRs and two backports.

Reply:
I pushed the nested try

Reply:
Thanks ❤️